lib/inputs/text-input.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var _ = require('underscore');
+var BaseInput = require('./base-input');
+var errors = require('../errors');
+
+const DEFAULT_VALUE = '';
+
+class TextInput extends BaseInput {
+    constructor(id, params) {
+        super('text', id, params);
+    }
+
+    getInputDefaultValue() {
+        return DEFAULT_VALUE;
+    }
+
+    normalizeParams(params) {
+        if (params.default) {
+            try {
+                params = Object.assign({}, params, {
+                    default: this.normalizeValue(params.default)
+                });
```

Comment:
I think `_.extend(_.clone(params), { default: ... })` would better capture what you are doing here.